polyfill/lib/zoneddatetime.mjs

```diff
@@ -128,7 +128,51 @@ export class ZonedDateTime {
   }
   get hoursInDay() {
     if (!ES.IsTemporalZonedDateTime(this)) throw new TypeError('invalid receiver');
-    throw new Error('hoursInDay not implemented yet');
+    const dt = dateTime(this);
+    const DateTime = GetIntrinsic('%Temporal.DateTime%');
+    const year = GetSlot(dt, ISO_YEAR);
+    const month = GetSlot(dt, ISO_MONTH);
+    const day = GetSlot(dt, ISO_DAY);
+    const today = new DateTime(year, month, day, 0, 0, 0, 0, 0, 0);
+    const tomorrowFields = ES.AddDate(year, month, day, 0, 0, 0, 1, 'reject');
+    const tomorrow = new DateTime(tomorrowFields.year, tomorrowFields.month, tomorrowFields.day, 0, 0, 0, 0, 0, 0);
+    const timeZone = GetSlot(this, TIME_ZONE);
+    const todayNs = GetSlot(ES.GetTemporalInstantFor(timeZone, today, 'compatible'), EPOCHNANOSECONDS);
+    const tomorrowNs = GetSlot(ES.GetTemporalInstantFor(timeZone, tomorrow, 'compatible'), EPOCHNANOSECONDS);
+    let { seconds, milliseconds, microseconds, nanoseconds } = ES.DifferenceInstant(
```

Comment:
Why is all the code below here needed? Could it be replaced with something like this?
```js
return Number(tomorrowNs - todayNs) / 3.6e12;
```

Reply:
Or is there some non-ISO time calendar reason to want to do it the way you did?

Reply:
Good point. We still do need to calculate the quotient and remainder separately because BigInt division is integer division, but this can be simplified a lot.

Reply:
Why not just convert the difference to a `Number` and then use floating-point `Number` math?  If a "day" is longer than 140 24-hour days, some precision loss seems OK. ;-)

Reply:
In the spec text, the epoch nanoseconds are mathematical values (ℝ values) so you have to be explicit about when you convert them to Number values, my guess is that this is the cleanest way.